From 4e3393cfceb338b6e777bfd44e51cbf4e4601291 Mon Sep 17 00:00:00 2001
From: Yu Yue <yuyue@mprc.pku.edu.cn>
Date: Mon, 31 Oct 2011 11:29:03 +0800
Subject: [PATCH 046/641] UniCore64: Add function cpu_idle

Fixed error:
noinitramfs.c:(.ref.text+0x74): undefined reference to `cpu_idle'

Signed-off-by: Yu Yue <yuyue@mprc.pku.edu.cn>
---
 arch/unicore64/kernel/process.c |   14 ++++++++++++++
 1 file changed, 14 insertions(+)

diff --git a/arch/unicore64/kernel/process.c b/arch/unicore64/kernel/process.c
index 59eba48..eb596e4 100644
--- a/arch/unicore64/kernel/process.c
+++ b/arch/unicore64/kernel/process.c
@@ -81,3 +81,17 @@ unsigned long get_wchan(struct task_struct *p)
 	BUG();
 	return 0;
 }
+
+/**
+ * cpu_idle() - The idle thread
+ *
+ * The idle thread. There's no useful work to be
+ * done, so just try to conserve power and have a
+ * low exit latency (ie sit in a loop waiting for
+ * somebody to say that they'd like to reschedule)
+ */
+void cpu_idle(void)
+{
+	/* FIXME */
+	BUG();
+}
-- 
1.7.9.5

